### ISSUE TITLE
Updating to v1 break the docusaurus sitemap

### ISSUE THREAD

**Issue by sean-ww** [rohit-gohri/redocusaurus#167]
I found moving from redocusaurus 0.4 to 1.0.2 caused our docusaurus sitemap to disappear. I couldn't see any errors, it's just we noticed it upon deployment. With docusaurus 2.0.0-beta.3 and 2.0.0-beta.18.

**Comment by rohit-gohri** [comment]
Are you sure the problem is because of redocusaurus and not because of the update to beta.18? Because the sitemap in the demo website still works: https://redocusaurus.vercel.app/sitemap.xml

Maybe there might be changes in config required to make it work?

**Comment by sean-ww** [comment]
I first investigated that but then rolled back to beta.3 and it was still the case. When keeping everything the same and only upgrading redocusaurus we lost our sitemap. When rolling back the redocusaurus version it was there again. We also use @docusaurus/preset-classic and @docusaurus/plugin-content-docs. I see you have beta.17 (https://github.com/rohit-gohri/redocusaurus/blob/main/website/package.json) assuming that is the demo website, but otherwise I can’t see why it didn’t work for us.
(The original reason we were updating was due to some css hiding the redoc api search results - but I found that issue with both versions. I see you disableSearch in your demo website. For instance prior to v1: .redocusaurus .menu-content div:first-child { display: none }. Then again something else in v1.)

**Comment by rohit-gohri** [comment]
There was an issue with relative urls that might have caused the issue. Could you try upgrading to latest beta.20 and redocusaurus 1.0.3? I'm still not 100% sure though what the issue might be, if you can share the repo that would help debug

**Comment by sean-ww** [comment]
We can't upgrade at the moment, but here's an example copy of our setup: https://github.com/sean-ww/redocusaurus-example

**Comment by rohit-gohri** [comment]
Thanks for the example. I will have a look soon

**Comment by rohit-gohri** [comment]
Was able to reproduce the issue with the example

**Comment by rohit-gohri** [comment]
This was due to `noIndex` being set to true in your config. Not related to redocusaurus.

From the docs (https://docusaurus.io/docs/2.0.0-beta.20/api/plugins/@docusaurus/plugin-sitemap#ignorePatterns):

> This plugin also respects some site config:
>
>    [noIndex](https://docusaurus.io/docs/2.0.0-beta.20/api/docusaurus-config#noIndex): results in no sitemap generated

**Comment by sean-ww** [comment]
@rohit-gohri sorry I don't understand your explanation. Our code is: `noIndex: process.env.ENV !== 'prod',`. We only want the sitemap on prod. The issue was that moving from redocusaurus 0.4 to 1.0.2 stopped the sitemap appearing on our production site. We haven't changed that line of code, and you can see we have a sitemap using 0.4: https://docs.raisenow.com/sitemap.xml

**Comment by rohit-gohri** [comment]
> We haven't changed that line of code, and you can see we have a sitemap using 0.4:
Redocusaurus has some docusaurus packages as dependencies, so upgrading it must have caused your docusaurus versions to upgrade too.

In the example repo the sitemap plugin is 2.0.0-beta.3  https://github.com/sean-ww/redocusaurus-example/blob/b6027dbba4b760a5c74c0a6afee958cf807d7716/yarn.lock#L2229-L2230

And in beta-19 docusaurus team added this change: https://github.com/facebook/docusaurus/pull/7143 
In the older version it probably ignored the `noIndex` setting.


I'm able to successfully generate the sitemap in your example repo with  `noIndex: false` and these dependencies:
```
    "@docusaurus/core": "^2.0.0-beta.21",
    "@docusaurus/preset-classic": "^2.0.0-beta.21",
    "redocusaurus": "^1.0.0",
```

> Our code is: noIndex: process.env.ENV !== 'prod',. We only want the sitemap on prod.

This code will disable the sitemap whenever `ENV` is not set (default undefined case), I'm guessing the env var is not set properly.

**Comment by rohit-gohri** [comment]
@sean-ww I have opened a PR with my changes: https://github.com/sean-ww/redocusaurus-example/pull/1

You can verify if you are able to create sitemap or not. If not then we will re-open this issue.

**Comment by sean-ww** [comment]
I've now updated our actual project (not the example github repo). It's deployed and working for us now thanks on redocusaurus v1 with 2.0.0-beta.21 (also having to update our node version).
I do think ENV has always been set for us. You mentioned the old version ignoring noIndex, and it does look like that was true. It had no impact on sitemap creation previously. With our old dependencies, when ENV was 'stage' we still got a sitemap. So we actually had `rm ./build/sitemap.xml` in our stage deployment script. Now without changing ENV, we no longer get the sitemap on that environment as desired and could remove the rm line.
Thanks for your help!
